### PR TITLE
Ensure no data is returned when a function has no return values

### PR DESCRIPTION
**Pull request by seanyoung**
Signed-off-by: Sean Young <sean@mess.org>

### CODE REVIEW

File: src/codegen/dispatch.rs [hunk 1]
```diff
@@ -256,10 +256,23 @@ fn add_dispatch_case(
                 data_len: zext_len,
             },
         );
+    } else {
+        // TODO: On Solana, We could elide setting the return data if this function calls no external functions
+        // are called and replace this with a simple Instr::Return.
```
Comment (LucasSte) [inline src/codegen/dispatch.rs:261]: This comment isn't clear.

**Review by LucasSte — CHANGES_REQUESTED** [review]

File: src/codegen/dispatch.rs (hunk 1 above)
Comment (seanyoung) [reply src/codegen/dispatch.rs]: Is this better?

File: src/codegen/dispatch.rs [hunk 2]
```diff
@@ -256,10 +256,27 @@ fn add_dispatch_case(
                 data_len: zext_len,
             },
         );
+    } else {
+        // TODO: On Solana, We could elide setting the return data if this function calls no external functions
+        // are called and replace this with a simple Instr::Return, which does not set any return data.
```
Comment (LucasSte) [inline src/codegen/dispatch.rs:261]:
```suggestion
        // TODO: On Solana, We could elide setting the return data if this function calls no external functions
        // and replace this with a simple Instr::Return, which does not set any return data.
```

**Review by LucasSte — APPROVED** [review]

File: src/codegen/dispatch.rs (hunk 1 above)
Comment (LucasSte) [reply src/codegen/dispatch.rs]: Yep

**Review by xermicus — APPROVED** [review]

**Pull request merged**